runner/app/utils/nvml_manager.py

```diff
@@ -0,0 +1,37 @@
+"""
+This module manages NVML (NVIDIA Management Library) initialization and shutdown,
+ensuring efficient resource management and improved performance for GPU operations.
+"""
+import pynvml
+import logging
+import atexit
+
+logger = logging.getLogger(__name__)
+
+class NVMLManager:
+    """A class to manage NVML initialization and shutdown."""
+    def __init__(self):
+        self._initialized = False
+        atexit.register(self.shutdown)
+
+    def initialize(self):
+        """Initialize NVML."""
+        if not self._initialized:
+            try:
+                pynvml.nvmlInit()
+                self._initialized = True
+                logger.info("NVML initialized successfully.")
+            except pynvml.NVMLError as e:
+                logger.error(f"Failed to initialize NVML: {e}")
+
+    def shutdown(self):
+        """Shutdown NVML."""
+        if self._initialized:
+            try:
+                pynvml.nvmlShutdown()
+                self._initialized = False
+                logger.info("NVML shutdown successfully.")
+            except pynvml.NVMLError as e:
+                logger.error(f"Failed to shutdown NVML: {e}")
+
+nvml_manager = NVMLManager()
```

Comment:
Does this live the whole lifespan? I would think it would be set in the lifespan function?

Reply:
The object is created here but it is initialised in the [main.py/lifespawn](https://github.com/livepeer/ai-worker/blob/511f08f26bd73c2a49ed1c6837d6a0058674a13b/runner/app/main.py#L19) method.

Reply:
👍